docs/how-to-guides/support-new-model.md

```diff
@@ -0,0 +1,688 @@
+# How to support a new model
+
+## How does Kernl optimize a model
+
+### Overview
+
+To optimize a model, Kernl uses [TorchDynamo](https://github.com/pytorch/torchdynamo) JIT compiler and provides a custom backend where we replace part of the [Torch FX](https://pytorch.org/docs/1.13/fx.html) graph with optimized kernels.
+
+The custom backend is defined in [src/kernl/model_optimization.py](https://github.com/ELS-RD/kernl/blob/v0.1.0/src/kernl/model_optimization.py#L44)
+
+``` { .py }
+def compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+    dynamo_backend_ofi(gm)
+    return cuda_graphs_wrapper(gm, example_inputs, pool=pool)
+```
+This backend combines two steps:
+
+- First one is to apply the graph replacements
+- Second one is to use [CUDA graphs](https://pytorch.org/blog/accelerating-pytorch-with-cuda-graphs/)
+
+The second step eliminates most of the CPU overhead but we won't elaborate on this and focus on the first one that does the graph replacements.
+
+
+### Inspecting the FX Graph
+
+First, a few words about [Torch FX](https://pytorch.org/docs/1.13/fx.html).
+Torch FX is a torch module to module transformation toolkit. It can trace a torch module (or a function) execution. All the operations are then recorded into a list of nodes that are represented as an FX Graph. From this graph, Torch FX generates python code matching the graph's semantics. Both graph and python code are accessible from the Torch FX GraphModule, which is also a torch module instance. Torch FX allows us to play with FX graphs but stay at the torch module level.
```

Comment:
into a list of nodes that are represented as an FX Graph -> into a graph of nodes

Reply:
Addressed in 7722ebd4ea1d643bfc1796d598711f4b56142692